platforms/jvm/plugins-java/src/main/java/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponent.java

```diff
@@ -78,100 +57,51 @@ public DefaultJvmSoftwareComponent(
 
     @Override
     public void withJavadocJar() {
-        mainFeature.withJavadocJar();
-
-        Configuration javadocElements = mainFeature.getJavadocElementsConfiguration();
-        if (!isRegisteredAsLegacyVariant(javadocElements)) {
-            addVariantsFromConfiguration(javadocElements, new JavaConfigurationVariantMapping("runtime", true));
-        }
+        getFeatures().configureEach(feature -> {
```

Comment:
🤔 I wonder how useful these methods on the component actually are now.  Seems to me most of the time you would want more granular control over what sources/javadocs are published if you had multiple features on your component.  I think accessing the specific feature (or the main feature) you are interested in and using the new methods on the feature is probably good enough.